rclcpp/test/test_function_traits.cpp

```diff
@@ -317,12 +299,122 @@ TEST(TestFunctionTraits, argument_types) {
 }
 
 /*
+   Tests that funcion_traits checks the types of the arguments of several functors.
+ */
+TEST(TestFunctionTraits, check_argument_types) {
+  // Test regular functions
+  static_assert(
+    rclcpp::check_argument_types<decltype(func_one_int), int>::value,
+    "Functor accepts a single int as arguments");
+
+  static_assert(
+    !rclcpp::check_argument_types<decltype(func_one_int), char>::value,
+    "Functor does not accept a char as argument");
+
+  static_assert(
+    !rclcpp::check_argument_types<decltype(func_one_int), char, int>::value,
+    "Functor does not accept two arguments");
+
+  static_assert(
+    !rclcpp::check_argument_types<decltype(func_two_ints), int>::value,
+    "Functor accepts two ints as arguments");
```

Comment:
The text above says:

    Functor does not accept two arguments

Since this test is similar I would expect a similar text.